packages/agoric-cli/src/commands/inter.js

```diff
@@ -153,57 +252,119 @@ For example:
     .action(async opts => {
-      const { agoricNames, readLatestHead } = await rpcTools();
+      const { agoricNames, readLatestHead } = await tryMakeUtils();
 
       const [metrics, quote] = await Promise.all([
         readLatestHead(`published.vaultFactory.manager${opts.manager}.metrics`),
         readLatestHead(`published.vaultFactory.manager${opts.manager}.quotes`),
       ]);
       const info = fmtMetrics(metrics, quote, values(agoricNames.vbankAsset));
-      stdout.write(JSON.stringify(info, bigintReplacer, 2));
-      stdout.write('\n');
+      show(info, true);
     });
 
   const bidCmd = interCmd
     .command('bid')
     .description('auction bidding commands');
 
-  bidCmd
-    .command('by-price')
-    .description('Print an offer to bid collateral by price.')
-    .requiredOption('--price [number]', 'bid price', Number)
-    .requiredOption('--giveCurrency [number]', 'Currency to give', Number)
-    .requiredOption(
-      '--wantCollateral [number]',
-      'Collateral expected for the currency',
-      Number,
-    )
-    .option('--collateralBrand [string]', 'Collateral brand key', 'IbcATOM')
-    .option('--offerId [number]', 'Offer id', String, `bid-${now()}`)
+  /**
+   * @param {string} from
+   * @param {import('@agoric/smart-wallet/src/offers.js').OfferSpec} offer
+   * @param {Awaited<ReturnType<tryMakeUtils>>} tools
+   */
+  const placeBid = async (from, offer, tools) => {
+    const { networkConfig, agoricNames, pollOffer } = tools;
+    const io = { ...networkConfig, execFileSync, delay, stdout };
+
+    const { home, keyringBackend: backend } = interCmd.opts();
+    const result = await sendAction(
+      { method: 'executeOffer', offer },
+      { keyring: { home, backend }, from, verbose: false, ...io },
+    );
+    const { timestamp, txhash, height } = result;
+    console.error('bid is broadcast:');
+    show({ timestamp, height, offerId: offer.id, txhash });
+    const found = await pollOffer(from, offer.id, height);
+    // TODO: command to wait 'till bid exits?
+    const bid = coerceBid(found, agoricNames, console.warn);
+    if (!bid) {
+      console.warn('malformed bid', found);
+      return;
+    }
+    const info = fmtBid(bid, values(agoricNames.vbankAsset));
+    show(info);
+  };
+
+  /** @param {string} literalOrName */
+  const normalizeAddress = literalOrName =>
+    normalizeAddressWithOptions(literalOrName, interCmd.opts(), {
+      execFileSync,
+    });
+
+  /**
+   * @typedef {{
+   *   give: string,
+   *   desiredBuy: string,
+   *   wantMinimum?: string,
+   *   offerId: string,
+   *   from: string,
+   *   generateOnly?: boolean,
+   * }} SharedBidOpts
+   */
+
+  /** @param {ReturnType<createCommand>} cmd */
+  const withSharedBidOptions = cmd =>
+    cmd
+      .requiredOption(
+        '--from <address>',
+        'wallet address literal or name',
+        normalizeAddress,
+      )
+      .requiredOption('--give <amount>', 'IST to bid')
+      .option(
```

Comment:
```suggestion
      .requiredOption(
```

Reply:
The `--desired-buy` option has a default -- a very high number (1 million, IIRC). So while it's required by the contract, the user doesn't have to make up some arbitrary high number each time.